src/app/clusters/testing/ClusterTester.h

```diff
@@ -229,7 +261,109 @@ class ClusterTester
         return mTestServerClusterContext.EventsGenerator().GetNextEvent();
     }
 
+    /// CHIP Test Cluster Write Helpers
+    //  - Multi-fabric writes are supported through WriteOperation subject descriptors.
+    // - Cluster paths are automatically deduced using cluster.GetPaths().
+
+    //  - Scalars/enums:
+    //    * Written directly using DecoderFor().
+    //    * `value` must match the attribute's ::DecodableType.
+    //    * Example: WriteAttribute(attrId, 42, fabricIndex);
+
+    //  - Structs/ZAP structs: use WriteAttribute(attrId, structValue, fabricIndex)
+    //     * Encoded into a TLV buffer via Encode() or EncodeForWrite().
+    //     * `value` must match ::DecodableType.
+    //     * Example: MyStruct s{42, true}; WriteAttribute(attrId, s, fabricIndex);
+
+    template <typename T>
+    CHIP_ERROR WriteAttribute(AttributeId attr, const T & value, FabricIndex fabricIndex)
+    {
+        const auto & paths = mCluster.GetPaths();
+        VerifyOrReturnError(paths.size() == 1u, CHIP_ERROR_INCORRECT_STATE);
+
+        app::ConcreteAttributePath path(paths[0].mEndpointId, paths[0].mClusterId, attr);
+        app::Testing::WriteOperation writeOp(path);
+
+        // Create a stable object on the stack beware of a dangling ptr fabric index will change
+        Access::SubjectDescriptor subjectDescriptor{ fabricIndex };
+        writeOp.SetSubjectDescriptor(subjectDescriptor);
+
+        uint8_t buffer[1024];
+        TLV::TLVWriter writer;
+        writer.Init(buffer);
+
+        ReturnErrorOnFailure(EncodeValueToTLV(writer, value));
+
+        TLV::TLVReader reader;
+        reader.Init(buffer, writer.GetLengthWritten());
+        ReturnErrorOnFailure(reader.Next());
+
+        app::AttributeValueDecoder decoder(reader, *writeOp.GetRequest().subjectDescriptor);
+        return mCluster.WriteAttribute(writeOp.GetRequest(), decoder).GetUnderlyingError();
+    }
+
+    // - Lists: use WriteAttribute(attrId, listValue, fabricIndex)
+    //     * Encoded as TLV arrays. Each element is encoded individually.
+    //         - Struct elements must provide EncodeForWrite().
+    //         - Simple elements (integral, enum) handled by generic templated encoding.
```

Comment:
simple elements should not be fabric scoped though.

I generally have some questions/concerns about the APIs:
  - we ask for a fabricIndex without validating that this applies to fabric scoped elements only
  - we seem to sepcial case List... can we avoid that? i believe DataModel::Encode tries to be generic so you always call `Encode` and `EncodeForWrite`. Can't we use that instead? Or could we EncodeForWrite always for fabric scoped and use Encode for non-fabric scoped. https://github.com/project-chip/connectedhomeip/blob/master/src/app/WriteClient.h#L390 seems to do this for `TryEncodeSingleAttributeDataIB` ... I imagine we want to match that.
  - we have long WriteAttribute implementations that have similar code ... can we deduplicate or remove one of them?